src/sentry/culprit.py

```diff
@@ -0,0 +1,72 @@
+"""
+This file implements the legacy culprit system.  The culprit at this point is
+just used as a fallback if no transaction is set.  When a transaction is set
+the culprit is overriden by the transaction value.
+
+Over time we want to fully phase out the culprit.  Until then this is the
+code that generates it.
+"""
+
+from __future__ import absolute_import
+from sentry.constants import MAX_CULPRIT_LENGTH
+from sentry.utils.safe import get_path
+from sentry.utils.strings import truncatechars
+
+
+def generate_culprit(data):
+    platform = data.get('platform')
+    exceptions = get_path(data, 'exception', 'values')
```

Comment:
```suggestion
    exceptions = get_path(data, 'exception', 'values', filter=True)
```
This would skip `null` exceptions that might be generated during normalization.